projects/valtimo/object-management/src/lib/components/object-management-detail-container/tabs/object-management-list-search-fields/object-management-list-search-fields.component.ts

```diff
@@ -0,0 +1,428 @@
+
+
+        this.formGroup.patchValue({
+          key: column.key,
+          title: column.title,
+          path: column.path,
+          // @ts-ignore
+          dataType: {...dataTypeItem},
+          // @ts-ignore
+          fieldType: {...fieldTypeItem},
+        });
+
+        this.openModal('edit');
+      });
+  }
+
+  private updateObjectManagementListSearchField(
+    objectId: string,
+    searchField: SearchField
+  ): void {
+
+    this.disableInput();
+    this.objectManagementService.putSearchField(objectId, this.formGroup.value.key, searchField).pipe(
+      tap(() => {
+        this.refreshObjectManagementListSearchFields();
+        localStorage.setItem(`list-search-fields${objectId}`, null);
+      }),
+      catchError(() => {
+        this.enableInput();
+        return of(null);
+      })
+    ).subscribe();
+  }
+
+  private addSearchField(): void {
+    const formValue = this.formGroup.value;
+
+    this.objectId$.pipe(
+      take(1),
+      switchMap(objectId => this.objectManagementService.postSearchField(objectId, this.mapFormValuesToColumn(formValue))),
+      tap(() => {
+        this.closeModal();
+        this.refreshObjectManagementListSearchFields();
+      }),
+      catchError(() => {
+        this.enableInput();
+        return of(null);
+      })
+    ).subscribe();
+  }
+
+  private updateSearchField(): void {
+    this.objectId$.pipe(
+      take(1),
+      switchMap(objectId => this.objectManagementService.putSearchField(objectId, this.formGroup.value.key, this.mapFormValuesToColumn(this.formGroup.value))),
+      tap(() => {
+        this.closeModal();
+        this.refreshObjectManagementListSearchFields();
+      }),
+      catchError(() => {
+        this.enableInput();
+        return of(null);
+      })
+    ).subscribe();
+  }
+
+  private setDownload(
+    objectId: string,
+    ObjectManagementSearchFields: Array<SearchField>
+  ): void {
+    this.downloadName$.next(`${objectId}.json`);
+    this.downloadUrl$.next(
+      this.sanitizer.bypassSecurityTrustUrl(
+        'data:text/json;charset=UTF-8,' +
+        encodeURIComponent(JSON.stringify(ObjectManagementSearchFields, null, 2))
+      )
+    );
+  }
+
+  private disableInput(): void {
+    this.disableInput$.next(true);
+    this.formGroup.disable();
+  }
+
+  private enableInput(): void {
+    this.disableInput$.next(false);
+    this.formGroup.enable();
+  }
+
+  private refreshObjectManagementListSearchFields(): void {
+    this.refreshObjectManagementListSearchFields$.next(null);
+  }
+
+  private resetFormGroup(): void {
+    this.formGroup.reset();
+    this.selectedDataTypeItemIndex$.next(0);
+    this.selectedFieldTypeItemIndex$.next(0);
+  }
+
+  private mapFormValuesToColumn(formValue: any): SearchField {
+    return {
+      key: formValue.key,
+      title: formValue.title || '',
+      path: formValue.path,
+      dataType: formValue.dataType.key,
+      fieldType: formValue.fieldType.key
```

Comment:
This gives an error: 'Missing matchType'

Butnew searchFields don't have a matchType

Reply:
You need to update the config package.

Reply:
I did a rebuild locally and it worked. But now I see that the build failed with the same error:
```
error TS2741: Property 'matchType' is missing in type '{ key: any; title: any; path: any; dataType: any; fieldType: any; }' but required in type 'SearchField'.
```

Reply:
I will take a look at it.

Reply:
I made a new commit which fixes this issue.